### PR TITLE
(odsp-client): Setup e2e test cases 

### PR DESCRIPTION
**Pull request by sonalideshpandemsft**
This PR establishes end-to-end test cases that mirror those in azure-client. While the test cases remain identical, the distinction lies in the client configuration and token provider used.

image

List of remaining/followups work:
1. Currently, the audience test has asserts commented, and the size of members always remains at 1, as a single M365 account is utilized for login. When a new test tenant is generated, these tests can be modified accordingly. [AB#6425](https://dev.azure.com/fluidframework/internal/_workitems/edit/6425)
2. Fix pending test (DONE)

[AB#6390](https://dev.azure.com/fluidframework/internal/_workitems/edit/6390)

### CODE REVIEW

File: packages/service-clients/end-to-end-tests/odsp-client/src/test/containerCreate.spec.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+	});
+
+	/**
+	 * Scenario: test if Odsp Client can get a non-exiting container.
+	 *
+	 * Expected behavior: an error should be thrown when trying to get a non-existent container.
+	 */
+	it("cannot load improperly created container (cannot load a non-existent container)", async () => {
+		const consoleErrorFn = console.error;
+		console.error = (): void => {};
```
Comment (anthony-murphy) [inline packages/service-clients/end-to-end-tests/odsp-client/src/test/containerCreate.spec.ts:140]: why does this touch the console. we should never use console

File: packages/service-clients/end-to-end-tests/odsp-client/package.json [hunk 2]
```diff
@@ -0,0 +1,99 @@
+{
+	"name": "@fluid-experimental/odsp-end-to-end-tests",
+	"version": "2.0.0-internal.7.4.0",
+	"private": true,
+	"description": "Odsp client end to end tests",
+	"homepage": "https://fluidframework.com",
+	"repository": {
+		"type": "git",
+		"url": "https://github.com/microsoft/FluidFramework.git",
+		"directory": "packages/service-clients/end-to-end-tests"
```
Comment (ChumpChief) [inline packages/service-clients/end-to-end-tests/odsp-client/package.json:10]:
```suggestion
		"directory": "packages/service-clients/end-to-end-tests/odsp-client"
```

**Review by ChumpChief — APPROVED** [review]
Makes sense - please also make sure there's a work item opened for getting the audience tests working with the extra tenant.

File: packages/service-clients/end-to-end-tests/odsp-client/src/test/containerCreate.spec.ts (hunk 1 above)
Comment (sonalideshpandemsft) [reply packages/service-clients/end-to-end-tests/odsp-client/src/test/containerCreate.spec.ts]: thanks for catching this. removed console

**Review by anthony-murphy — APPROVED** [review]

**Pull request merged**